src/main/java/net/glowstone/entity/GlowPlayer.java

```diff
@@ -2826,10 +2846,47 @@ public GlowBlock getDigging() {
         return digging;
     }
 
+    private void broadcastBlockBreakAnimation(GlowBlock block, int destroyStage) {
+        if (block == null) return;
+
+        GlowChunk.Key key = new GlowChunk.Key(block.getChunk().getX(), block.getChunk().getZ());
+        // TODO: should canSeeChunk filtering take place here, in GlowPlayer, or both?
+        block.getWorld().getRawPlayers().stream().filter(player1 -> player1.canSeeChunk(key)).forEach(player2 -> {
```

Comment:
Ok. Why did you name your variables player1 and player2? Can't they both be player?

Also you have an if statement within the for-each lambda that could be part of the filter predicate.